### PR TITLE
Fixes the "nonzero is not supported for tensors with more than INT_MAX elements" in get_all_triplets_indices

### PR DESCRIPTION
**Pull request by mkmenta**
This solves #688

### CODE REVIEW

**Comment by mkmenta** [comment]
I am not sure if there could be other places where changing the `torch.where` could be beneficial (in case this implementation is accepted).

I think that in `get_all_pairs_indices` it would be hard to have a tensor with `INT_MAX` elements because the tensors used in the `torch.where` in this case are just `len(labels)*len(ref_labels)`.

**Comment by KevinMusgrave** [comment]
Thanks!
